node_modules/gh-orgunit/lib/internal/import.js

```diff
@@ -0,0 +1,594 @@
+    // Perform basic input validation
+    _validateOrgUnitMetadata(validator, orgUnit, parent, _currentLevel, _seenTypes);
+
+    // Ensure the external id for this organisational unit is unique within the tree
+    if (parent || orgUnit.externalId) {
+        validator.check(_seenExternalIds[orgUnit.externalId], {'code': 400, 'msg': util.format('Duplicate externalId found: %s', orgUnit.externalId)}).isNull();
+        _seenExternalIds[orgUnit.externalId] = orgUnit.displayName;
+    }
+
+    // Validate each child organisational unit
+    _.each(orgUnit.children, function(childOrgUnit) {
+        _validateImportOrgUnit(validator, childOrgUnit, orgUnit, _currentLevel + 1, _seenExternalIds, _seenTypes);
+    });
+
+    // Validate each series within this organisational unit
+    _.each(orgUnit.series, function(series) {
+        _validateImportSeries(validator, series, _seenExternalIds);
+    });
+
+    // TODO: Allow to import events directly associated to an organisational unit
+};
+
+/**
+ * Validate the metadata for an organisational unit
+ *
+ * @param  {Validator}      validator           The validator to validate each node with
+ * @param  {Object}         orgUnit             The organisational unit to validate
+ * @param  {Object}         [parent]        The parent of the given `orgUnit`
+ * @param  {Number}         currentLevel        The level on which the organisational unit appears in the entire tree
+ * @param  {Object}         seenTypes           A hash that maps previously seen organisational types to the level they were seen on
+ * @api private
+ */
+var _validateOrgUnitMetadata = function(validator, orgUnit, parent, currentLevel, seenTypes) {
+    validator.check(orgUnit.displayName, {'code': 400, 'msg': 'A valid displayName must be provided'}).notEmpty();
+    validator.check(orgUnit.displayName, {'code': 400, 'msg': 'A displayName can be at most 255 characters long'}).isShortString();
+
+    var validTypes = ['course', 'subject', 'part', 'module'];
+    validator.check(orgUnit.type, {'code': 400, 'msg': 'A valid type must be provided'}).notEmpty();
+    validator.check(orgUnit.type, {'code': 400, 'msg': 'A valid type must be one of: ' + validTypes.join(', ')}).isIn(validTypes);
+    // Ensure the type for this organisational unit is not on a different level
+    var typeLevel = seenTypes[orgUnit.type] || currentLevel;
+    validator.check(currentLevel, {'code': 400, 'msg': util.format('The type for %s was used on a different level', orgUnit.externalId)}).is(typeLevel);
+    seenTypes[orgUnit.type] = currentLevel;
+
+    if (orgUnit.description) {
+        validator.check(orgUnit.description, {'code': 400, 'msg': 'A description can be at most 1000 characters long'}).isMediumString();
+    }
+    if (orgUnit.metadata) {
+        if (_.isString(orgUnit.metadata)) {
+            try {
+                orgUnit.metadata = JSON.parse(orgUnit.metadata);
+            } catch (e) {
```

Comment:
We should keep some indication at least for support purposes of what went wrong (stack trace). Probably even pass the `e.message` into the result, especially if it gives more information about where parsing failed.